src/store/charts/mutations.ts

```diff
@@ -43,9 +43,8 @@ export const mutations: MutationTree<ChartState> = {
     }
     state[payload.type].push(payload.data)
     // console.log('set data', payload.type, payload.data)
-    while (state[payload.type].length > payload.retention) {
-      state[payload.type].splice(0, 1)
-    }
+    const firstInRange = state[payload.type].findIndex((entry: ChartData) => (Date.now() - entry.date.valueOf()) / 1000 < payload.retention)
+    if (firstInRange) state[payload.type].splice(0, firstInRange)
```

Comment:
@matmen I think this might have introduced a bug here, as `findIndex` can return -1 and will pass the test! One probably wants `> 0`, no?

Reply:
True! This doesn't change anything functionality wise (`.splice(0, -1)` doesn't mutate the array), but the extra call slows down execution very very slightly. I'll PR this together with marking the feature as beta